functional_tests.go

```diff
@@ -283,57 +367,70 @@ func testPutObjectReadAt() {
 
```

Comment:
`args["reader"]` needs to be populated with `reader` value, since `reader` is one of the parameters for `PutObject` function.

Reply:
done